drag_and_drop_v2/public/js/drag_and_drop.js

```diff
@@ -321,13 +321,73 @@ function DragAndDropTemplates(configuration) {
         )
     };
 
-    var mainTemplate = function(ctx) {
-        var problemTitle = ctx.show_title ? h('h2.problem-title', {innerHTML: ctx.title_html}) : null;
-        var problemHeader = ctx.show_problem_header ? h('h3.title1', gettext('Problem')) : null;
-        var popupSelector = 'div.popup';
-        if (ctx.popup_html && !ctx.last_action_correct) {
+    var itemFeedbackPopupTemplate = function(ctx) {
+        var popupSelector = 'div.popup.item-feedback-popup';
+        var msgs = ctx.feedback_messages || [];
+        var have_messages = msgs.length > 0;
+        var popup_content;
+
+        var close_button_describedby_id = "close-popup-"+configuration.url_name;
+
+        if (msgs.length > 0 && !ctx.last_action_correct) {
             popupSelector += '.popup-incorrect';
         }
+
+        if (ctx.mode == DragAndDropBlock.ASSESSMENT_MODE) {
+            var content_items = [
+                (!ctx.last_action_correct) ? h("p", {}, gettext("Some of your answers were not correct")) : null,
+                h("p", {}, gettext("Hints:")),
+                h("ul", {}, msgs.map(function(message) {
+                    return h("li", {innerHTML: message.message});
+                }))
+            ];
+            popup_content = h("div.popup-content", {}, have_messages ? content_items : []);
+        } else {
+            popup_content = h("div.popup-content", {}, msgs.map(function(message) {
+                return h("p", {innerHTML: message.message});
+            }))
+        }
+
+        return h(
+            popupSelector,
+            {
+                style: {display: have_messages ? 'block' : 'none'},
+                attributes: {"tabindex": "-1"}
+            },
+            [
+                h(
+                    'button.unbutton.close',
+                    {
+                        attributes: {
+                            'aria-hidden': !have_messages
```

Comment:
@haikuginger I know you didn't add this, but I'm wondering why this `button` needs to have an `aria-hidden` attribute: As per L354 above, if `have_messages` is false/falsy, we don't display the popup at all, so specifying that the close button should be hidden seems redundant?